src/main/services/csfloat/csfloatSocket.ts

```diff
@@ -0,0 +1,515 @@
+  ): this;
+}
+
+export class CSFloatSocket extends EventEmitter {
+  private connected: boolean;
+  private _csFloatClient: CSFloatClient;
+  private steamIDBase64: string;
+  private statusExtension = false;
+
+  constructor(_CSFloatClient: CSFloatClient, steamIDBase64: string) {
+    super();
+    this._csFloatClient = _CSFloatClient;
+    this.steamIDBase64 = steamIDBase64;
+  }
+
+  public disconnect() {
+    this.connected = false;
+    this.emit("stateChange", false);
+    return;
+  }
+
+  public async connect(): Promise<void> {
+    this.connected = true;
+    await this.registerLoops();
+  }
+
+  private async registerLoops(): Promise<void> {
+    while (this.connected) {
+      try {
+        this.emit("stateChange", true);
+        const tradeOffers: IGetTradeOffersResponde = await new Promise(
+          (resolve, reject) => {
+            this.emit(
+              "getSentTradeOffers",
+              async (getTradeOffersResponse: IGetTradeOffersResponde) => {
+                if (!getTradeOffersResponse) { reject(new Error("Failed to get trade offers")); }
+                resolve(getTradeOffersResponse);
+              }
+            );
+          }
+        );
+        const tradesInQueue = await this._csFloatClient.getTrades(
+          EStatusTradeCSFLOAT.QUEUED
+        );
+
+        await this.autoAcceptTrades(tradesInQueue, tradeOffers);
+
+        const tradesInPending = await this._csFloatClient.getTrades(
+          EStatusTradeCSFLOAT.PENDING
+        );
+
+        await this.verificationsLoop(tradesInPending);
+        await this.sendOffer(tradesInPending, tradeOffers);
+        await this.extensionLoop(
+          tradesInPending,
+          tradeOffers
+        );
+      } catch (err) {
+        this.emit("error", err);
+      }
+      await sleepAsync(minutesToMS(3));
+    }
+
+    this.disconnect();
+  }
+
+  private async extensionLoop(
+    tradesInPending: ITradeFloat[],
+    tradeOffers: IGetTradeOffersResponde
+  ): Promise<void> {
+    let errors: IUpdateErrors;
+
+    if (tradesInPending.length > 0) {
+      const ignoredOrBlokedUsers: string[] = await new Promise((resolve, reject) => {
+        this.emit(
+          "getBlockerUsers",
+          async (ignoredOrBlokedUsersResponse: string[]) => {
+            if (!ignoredOrBlokedUsersResponse) {
+              reject(new Error("Failed to get ignored and blocked users."));
+              return;
+            }
+            resolve(ignoredOrBlokedUsersResponse);
+          }
+        );
+      })
+
+      errors = await this.pingUpdates(
+        tradesInPending,
+        ignoredOrBlokedUsers,
+        tradeOffers
+      );
+    }
+
+    try {
+      this.statusExtension = await this._csFloatClient.pingExtensionStatus(
+        errors,
+        this.steamIDBase64
+      );
+      this.emit("stateChange", this.statusExtension);
+    } catch (error) {
+      console.error("failed to ping extension status to csfloat", error);
+    }
```

Comment:
Let the error be catched by the parent function
```suggestion
    this.statusExtension = await this._csFloatClient.pingExtensionStatus(
      errors,
      this.steamIDBase64
    );
    this.emit("stateChange", this.statusExtension);
```